extras/speaker-recognition/src/simple_speaker_recognition/services/transcription/deepgram.py

```
@@ -0,0 +1,254 @@
+        }
+
+        params = self._build_params(config)
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(
+                url, headers=headers, data=audio_data, params=params
+            ) as response:
+                if response.status != 200:
+                    error_text = await response.text()
+                    log.error(f"Deepgram API error: {response.status} - {error_text}")
+                    raise Exception(f"Deepgram API error: {error_text}")
```

Ruff (0.14.8)

Create your own exception

(TRY002)